src/shogun/regression/GLM.cpp

```diff
@@ -0,0 +1,82 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/regression/GLM.h>
+#include <utility>
+using namespace shogun;
+
+GLM::GLM() : LinearMachine()
+{
+	init();
+}
+SGVector<float64_t> log_likelihood(
+    const std::shared_ptr<DenseFeatures<float64_t>> features,
+    std::shared_ptr<Labels> label)
+{
+	int32_t vector_count = features->get_num_vectors();
+	int32_t feature_count = features->get_num_features();
+	ASSERT(vector_count > 0 && label->get_num_labels() == vector_count)
+	// Array of Lambdas
+	SGVector<float64_t> lambda(vector_count);
+	for (int32_t i = 0; i < vector_count; i++)
+	{
+		SGVector<float64_t> feature_vector = features->get_feature_vector(i);
```

Comment:
auto

Reply:
@Khalifa1997 why do you resolve conversations when you haven't addressed the comment by @vigsterkr ?

Reply:
he was mentionned auto to be used in the for loop and I think I had address them all in my code +he mentionned the ctor thing when it took default values. I fixed this by giving default values inline with member attributes. but if I had missed any comment, then yea I appologise and ofc u can point it out :)

Reply:
You can always use auto if possible. The only case where I would write the type is if it would be unclear/confusing otherwise :)